src/app/content/experiments/reducer.ts

```diff
@@ -0,0 +1,25 @@
+import { Reducer } from 'redux';
+import { getType } from 'typesafe-actions';
+import { receiveFeatureFlags } from '../../actions';
+import { AnyAction } from '../../types';
+import { experimentIds } from './constants';
+import { State } from './types';
+
+export const initialState: State = {};
+
+const reducer: Reducer<State, AnyAction> = (state = initialState, action): any => {
+    switch (action.type) {
+        case getType(receiveFeatureFlags):
+            const [id, variant] = action.payload;
+            if (experimentIds[id]) {
+                const experiment = experimentIds[id];
+                const idx = parseInt(variant, 10);
+                return {...state, [experiment]: idx};
+            }
```

Comment:
@TomWoodward imagine you will have thoughts on how to better organize this. [here is how the store looks.](https://share.getcloudapp.com/4gu2glWG)

Reply:
if you're gonna have it doing processing of the experiment id/mapping stuff, i'd put that behind an action named more specifically just for the experiments, because feature flags could also indicate the release flags and there is no reason for them to go through the same logic

i also think it would be neat for the value to be mapped before going into the state, so for debugging it will have a more human readable value

Reply:
done. makes sense.